### ISSUE TITLE
Allow knitrBootstrap to be used as a vignette engine

### ISSUE THREAD

**Issue by dtenenba** [jimhester/knitrBootstrap#169]
Currently if a package vignette wants to use knitrBootstrap it has to have a Makefile.
However, if knitrBootstrap registered a vignette engine, then the package author would just have to specify `VignetteEngine: knitrBootstrap` in `DESCRIPTION` and `%\VignetteEngine{knitrBootstrap::knit_bootstrap}` in the vignette. See the [discussion](http://cran.r-project.org/doc/manuals/r-release/R-exts.html#Non_002dSweave-vignettes) on non-Sweave vignettes in Writing R-extensions, and the help for `?tools::vignetteEngine`.

**Comment by jimhester** [comment]
This is true.  Pull request #68 should be able to do this using the rmarkdown vignetteEngine already, but it has some other unresolved issues as well.  Unfortunately I have been working on other side-projects so knitrBootstrap has been sadly neglected.

Rmarkdown now has ~95% of the features of knitrBootstrap, so I am not sure how much further it is worth taking this project, it seems mostly redundant to me at this point.

**Comment by jimhester** [comment]
That being said, registering a vignetteEngine shouldn't be too much work, so it can probably be done.

**Comment by dtenenba** [comment]
Thanks for the feedback. I am dealing with a package that is using knitrBootstrap and having issues, maybe I'll see if the maintainer wants to switch to rmarkdown and/or ask what specific features of knitrBootstrap he wants.
